src/plugins/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -452,7 +458,54 @@ export class SearchInterceptor {
           body: JSON.stringify({
             ...request,
             ...searchOptions,
+            stream:
+              strategy === ESQL_ASYNC_SEARCH_STRATEGY || strategy === ENHANCED_ES_SEARCH_STRATEGY,
```

Comment:
I think `strategy` is optional and defaults currently to `ENHANCED_ES_SEARCH_STRATEGY` on the server if it isn't provided, so maybe we can include that condition here as well:
```suggestion
              strategy === ESQL_ASYNC_SEARCH_STRATEGY || strategy === ENHANCED_ES_SEARCH_STRATEGY || strategy === undefined, // undefined strategy is treated as enhanced ES
```